packages/ra-core/src/dataProvider/useInfiniteGetList.ts

```diff
@@ -97,50 +107,58 @@ export const useInfiniteGetList = <RecordType extends RaRecord = any>(
                     pageParam,
                     pageInfo,
                 })),
-        {
-            onSuccess: data => {
-                // optimistically populate the getOne cache
-                data.pages.forEach(page => {
-                    page.data.forEach(record => {
-                        queryClient.setQueryData(
-                            [
-                                resource,
-                                'getOne',
-                                { id: String(record.id), meta },
-                            ],
-                            oldRecord => oldRecord ?? record
-                        );
-                    });
-                });
-            },
-            ...options,
-            getNextPageParam: lastLoadedPage => {
-                if (lastLoadedPage.pageInfo) {
-                    return lastLoadedPage.pageInfo.hasNextPage
-                        ? lastLoadedPage.pageParam + 1
-                        : undefined;
-                }
-                const totalPages = Math.ceil(
-                    (lastLoadedPage.total || 0) / pagination.perPage
-                );
+        initialPageParam: pagination.page,
+        ...queryOptions,
+        getNextPageParam: lastLoadedPage => {
+            if (lastLoadedPage.pageInfo) {
+                return lastLoadedPage.pageInfo.hasNextPage
+                    ? lastLoadedPage.pageParam + 1
+                    : undefined;
+            }
+            const totalPages = Math.ceil(
+                (lastLoadedPage.total || 0) / pagination.perPage
+            );
 
-                return lastLoadedPage.pageParam < totalPages
-                    ? Number(lastLoadedPage.pageParam) + 1
+            return lastLoadedPage.pageParam < totalPages
+                ? Number(lastLoadedPage.pageParam) + 1
+                : undefined;
+        },
+        getPreviousPageParam: lastLoadedPage => {
+            if (lastLoadedPage.pageInfo) {
+                return lastLoadedPage.pageInfo.hasPreviousPage
+                    ? lastLoadedPage.pageParam - 1
                     : undefined;
-            },
-            getPreviousPageParam: lastLoadedPage => {
-                if (lastLoadedPage.pageInfo) {
-                    return lastLoadedPage.pageInfo.hasPreviousPage
-                        ? lastLoadedPage.pageParam - 1
-                        : undefined;
-                }
+            }
+
+            return lastLoadedPage.pageParam === 1
+                ? undefined
+                : lastLoadedPage.pageParam - 1;
+        },
+    });
+
+    useEffect(() => {
+        if (result.data != null) {
+            // optimistically populate the getOne cache
+            result.data.pages.forEach(page => {
```

Comment:
I'm thinking we should put the same limit as in `useGetList` to avoid slowing down the UI too much with long lists.

Reply:
Another PR